sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -383,4 +426,21 @@ static inline bool _node_unreachable(_nib_onl_entry_t *node)
     }
 }
 
+uint32_t _evtimer_lookup(const void *ctx, uint16_t type)
+{
+    evtimer_msg_event_t *event = (evtimer_msg_event_t *)_nib_evtimer.events;
+    uint32_t offset = 0;
+
+    DEBUG("nib: lookup ctx = %p, type = %u\n", (void *)ctx, type);
+    while (event != NULL) {
+        offset += event->event.offset;
```

Comment:
Why is this adding up the offsets of *all* events up until the matching one?
What is this function doing?

Reply:
Because the offset of an event is referring to the event before, not the absolute offset.